cirq-core/cirq/work/observable_measurement.py

```diff
@@ -49,6 +51,9 @@
 if TYPE_CHECKING:
     import cirq
     from cirq.value.product_state import _NamedOneQubitState
+    from dataclasses import dataclass as json_serializable_dataclass
+else:
+    from cirq.protocols import json_serializable_dataclass
```

Comment:
Does the Cirq `json_serializable_dataclass` break type checking? Using two different classes under the same name feels really unstable to me.

Reply:
It breaks type checking. I introduced #4391 to work around this, but it isn't merged yet.

Reply:
Got it. I stand by my comment, though - this should wait until #4391 is merged and use the fix it provides.

Reply:
Ok. We definitely have this pattern in Cirq already as the linked pr demonstrates. It's really a drawback of mypy more than anything else; the decorator works fine